src/ORM/Table.php

```diff
@@ -934,11 +934,11 @@ public function hasBehavior($name)
     }
 
     /**
-     * Returns an association object configured for the specified alias if any
+     * Returns an association object configured for the specified alias if any.
      *
      * @deprecated 3.6.0 Use getAssociation() instead.
      * @param string $name the alias used for the association.
-     * @return \Cake\ORM\Association|null Either the association or null.
+     * @return \Cake\ORM\Association The association.
```

Comment:
I don't see the benefit of changing return type of deprecated method. Should be done just for `getAssociation()`.

Reply:
That was an oversight, my bad.